Sources/Client/Utils/NestableCodingKey.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  NestableCodingKey.swift
+//  StreamChatClient
+//
+//  Created by Bahadir Oncel on 1.04.2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+// Courtesy of AliSoftware
+// https://gist.github.com/AliSoftware/89b275d7259d23ebf12d377b6ffe15cd
+
+import Foundation
+
+/// Use this to annotate the properties that require a depth traversal during decoding.
+/// The corresponding `CodingKey` for this property must be a `NestableCodingKey`
+@propertyWrapper
+public struct NestedKey<T: Decodable>: Decodable {
+    struct AnyCodingKey: CodingKey {
+        let stringValue: String
+        let intValue: Int?
+        
+        init(stringValue: String) {
+            self.stringValue = stringValue
+            self.intValue = nil
+        }
+        
+        init?(intValue: Int) {
+            self.stringValue = "\(intValue)"
+            self.intValue = intValue
+        }
+    }
+    
+    public let wrappedValue: T?
+    
+    public init(from decoder: Decoder) throws {
+        guard let key = decoder.codingPath.last else {
+            throw DecodingError.valueNotFound(CodingKey.self,
+                                              DecodingError.Context(codingPath: decoder.codingPath,
+                                                                    debugDescription: "No CodingKey found in codingPath"))
+        }
```

Comment:
```suggestion
        guard let key = decoder.codingPath.last else {
            let context = DecodingError.Context(codingPath: decoder.codingPath, debugDescription: "No CodingKey found in codingPath")
            throw DecodingError.valueNotFound(CodingKey.self, context)
        }
```

Reply:
Why would we need that? We won't reuse context, and it's easy to read

Reply:
it's only for easy to read